Dockerfile

```diff
@@ -7,8 +7,17 @@ FROM python:3.5.2-alpine
 
 WORKDIR azure-cli
 COPY . /azure-cli
-RUN pip install --upgrade pip wheel
-RUN apk update && apk add bash gcc openssl-dev libffi-dev musl-dev
+# pip wheel - required for CLI packaging
+# jmespath-terminal - we include jpterm as a useful tool
+RUN pip install --upgrade pip wheel jmespath-terminal
+# bash gcc openssl-dev libffi-dev musl-dev - dependencies required for CLI
+# jq - we include jq as a useful tool
+# openssh - included for ssh-keygen
+# ca-certificates 
+# wget - required for installing jp
+RUN apk update && apk add bash gcc openssl-dev libffi-dev musl-dev jq openssh ca-certificates wget openssl && update-ca-certificates
+# We also, install jp
+RUN wget https://github.com/jmespath/jp/releases/download/0.1.2/jp-linux-amd64 -qO /usr/local/bin/jp && chmod +x /usr/local/bin/jp
```

Comment:
The docker image should be as small as possible. Installing `jp` not only increases the image size, but also causes legal issue. We should give the user the freedom to choose `jq`, `jp` or other tools.

Reply:
Installing `jp` from `https://github.com/` rather than the official package manager `apk` needs additional legal-related work: https://docs.opensource.microsoft.com/legal/cela-guidance/containers/

It also increases the image size. We should give the user the freedom to choose `jq`, `jp` or other tools.